lib/Interpreter/InterOp.cpp

```diff
@@ -2184,5 +2184,35 @@ namespace InterOp {
     }
     return dims;
   }
+
+  bool IsSubType(TCppType_t derived, TCppType_t base)
+  {
+    auto derivedType = clang::QualType::getFromOpaquePtr(derived);
+    auto baseType = clang::QualType::getFromOpaquePtr(base);
+
```

Comment:
I think we could implement that using `Sema::isDerivedFrom`: https://clang.llvm.org/doxygen/classclang_1_1Sema.html#ae9fa2ea2b4ce733c9f50241d25c45132

Reply:
@sudo-panda, what should that function return if we ask a nested type if it is a subtype?

Reply:
The input can be assumed to be a `RecordType`

Reply:
Yeah I agree we are looking for something like "IsTypeDerivedFrom". 

I didn't know that inner classes are also referred to as subtypes.

Reply:
See my initial comment, this is what we need to do, and change the interface name as @sudo-panda suggested.

Reply:
@Smit1603 this is not yet done